posts/2024-12-22-microprofile-7.adoc

```diff
@@ -0,0 +1,379 @@
+* Added a new `RestClientBuilder.baseUri(String)` overload method so users don't have to call `URI.create(String)`.
+* Added clarification in the spec on how to use Jakarta RESTful Web Services `EntityPart` API for multipart requests.
+
+=== Learn more
+
+For more information, see the following resources:
+
+* link:https://github.com/eclipse/microprofile-rest-client[Rest Client for MicroProfile] on GitHub
+* link:https://download.eclipse.org/microprofile/microprofile-rest-client-4.0/microprofile-rest-client-spec-4.0.html[Rest Client for MicroProfile 4.0 specification document]
+* link:http://download.eclipse.org/microprofile/microprofile-rest-client-4.0/apidocs/[API Javadoc]
+
+
+[#SUB_TAG_1]
+== MicroProfile Telemetry 2.0 (Logs, Metrics and Tracing)
+
+The link:{url-prefix}/docs/latest/reference/feature/mpTelemetry-2.0.html[MicroProfile Telemetry 2.0] (implemented by the Liberty feature `mpTelemetry-2.0`) helps improve the observability of your Java applications by using OpenTelemetry to collect and export logs, metrics, and traces in a standardized way. Previous versions (1.0, 1.1) of MicroProfile Telemetry could manage only distributed tracing.
+
+MicroProfile Telemetry 2.0 provides the latest OpenTelemetry technology. In addition to link:{url-prefix}/docs/latest/microprofile-telemetry.html[distributed tracing], the feature now allows OpenTelemetry to collect and export metrics and logs.
+For information about managing <<metrics, metrics>> and <<logs, logs>> with MicroProfile Telemetry, see the following sections.
+
+[#metrics]
+=== Send Liberty metrics to OpenTelemetry
+
+When the MicroProfile Telemetry 2.0 feature (`mpTelemetry-2.0`) is enabled, Open Liberty can now forward runtime component statistics that are captured by the link:{url-prefix}/docs/latest/reference/feature/monitor-1.0.html[Performance Monitoring 1.0] feature (`monitor-1.0`) to the MicroProfile Telemetry 2.0 runtime. This statistical data is registered as metrics in the telemetry runtime and can be forwarded to any OpenTelemetry Protocol (OTLP) compatible metric consumer to meet your monitoring needs.
+
+The following runtime components are supported:
+
+* `ThreadPool`
+* `Sessions`
+* `RequestTiming`
+* `ConnectionPool`
+
+To collect and export metrics, enable OpenTelemetry by using the following system property or environment variable:
+
+* System property: `otel.sdk.disabled=false`
+* Environment variable: `OTEL_SDK_DISABLED=false`
+
+You can set configuration properties in any of the link:{url-prefix}/docs/latest/external-configuration.html#default[config sources that are available to MicroProfile Config].
+
+[#manualMetrics]
+=== Use the OpenTelemetry API to mannually create metrics
+
+You can use the OpenTelemetry metrics API to define custom metrics in your application code.
+
+Before instrumenting your code, the following prerequisite is required:
+
+* Third-party APIs must be made visible for your application in the 
```

Comment:
do you also need to Add the opentelemetry API and OpenTelemetry instrumentation annotations as a provided dependency to your build path? eg https://docs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/docs/latest/reference/feature/mpTelemetry-2.0.html#dev

Reply:
no